src/aiokatcp/sensor.py

```diff
@@ -714,3 +732,114 @@ def copy(self) -> Dict[str, Sensor]:
     values.__doc__ = dict.values.__doc__
     items.__doc__ = dict.items.__doc__
     copy.__doc__ = dict.copy.__doc__
+
+
+class AggregateSensor(Sensor, metaclass=ABCMeta):
+    """A Sensor with its reading determined by several other Sensors.
+
+    This is an abstract class: the user must implement :meth:`update_aggregate`.
+    This method is called whenever the target :class:`SensorSet` has a sensor
+    added, removed, or one of the sensors changes its reading. The user-defined
+    :meth:`update_aggregate` returns the new reading for the AggregateSensor.
+
+    Parameters are all as per :class:`Sensor`, with the exception of
+    `target`, which is the :class:`SensorSet` from which the aggregated sensor
+    will determine its own reading.
+
+    Attributes
+    ----------
+    target : SensorSet
+        The set of sensors which will determine the reading of the aggregate
+        one. The aggregate sensor may be included in the set (e.g. in
+        `self.sensors` of a server) but it will not affect its own value.
+    """
+
+    def __init__(
+        self,
+        target: SensorSet,
+        sensor_type: Type[_T],
+        name: str,
+        description: str = "",
+        units: str = "",
+        *,
+        status_func: Callable[[_T], Sensor.Status] = _default_status_func,
```

Comment:
Can probably eliminate `status_func`, because `update_aggregate` always explicitly determines the status.